btree.go

```diff
@@ -64,6 +64,51 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+	grow     bool
+}
+
+func NewFreeList() *FreeList {
+	return newFreeList(true)
+}
+
+func newDefaultFreeList() *FreeList {
+	return newFreeList(false)
+}
+
+func newFreeList(grow bool) *FreeList {
```

Comment:
There is something to be said about simple API. I am willing to run experiments to see how big my freelist actually gets to see if it stays < 32 list. If it stays small enough, I'd be willing to keep the original bounded freelist.